### DISCUSSION TITLE
Compilation

### DISCUSSION THREAD

**Discussion by ahsadeghi** [pybamm-team/PyBaMM#3054] [Q&A]
Hello,
I am looking around to learn how to change the source code and compile it. Is there any tutorial for this?
if not, could you please explain how I can do it?

BR,
Hosein

**Comment by agriyakhetarpal** [comment]
Hi there, you may follow the documentation for installing from source available at https://docs.pybamm.org/en/latest/source/user_guide/installation/install-from-source.html/
and use the `-e` command-line argument with `pip` in order to install PyBaMM in editable mode. 

The documentation also offers information on how to compile the build-time requirements (the ODE and DAE solver components).

**Reply by ahsadeghi** [reply]
Thank you. Should I uninstall my regular Pybamm first, or can I have both?

**Reply by agriyakhetarpal** [reply]
You can have both of them, they just need to be in separate virtual environments
